lib/workspace.js

```diff
@@ -18,38 +19,36 @@ const sql = sqlLoader.loadSqlEquiv(__filename);
 
 const zipPrefix = config.workspaceGradedFilesReceiveDirectory;
 
-// https://stackoverflow.com/a/46213474/13138364
-const s3Sync = function (s3Path, bucketName) {
+async function uploadDirectoryToS3Async(s3Path, bucketName) {
     const s3 = new AWS.S3();
 
-    function walkSync(currentDirPath, callback) {
-        fs.readdirSync(currentDirPath).forEach(function (name) {
-            const filePath = path.join(currentDirPath, name);
-            const stat = fs.statSync(filePath);
+    async function walkDirectory(directory) {
+        const files = await fsPromises.readdir(directory);
+        await async.each(files, async (file) => {
+            const filePath = path.join(directory, file);
+            const stat = await fsPromises.stat(filePath);
             if (stat.isFile()) {
-                callback(filePath, stat);
-            } else if (stat.isDirectory()) {
-                walkSync(filePath, callback);
+                const fileBody = await fsPromises.readFile(filePath);
+                const bucketPath = filePath.substring(s3Path.length + 1);
+                const params = {
+                    Bucket: bucketName,
+                    Key: bucketPath,
+                    Body: fileBody,
+                };
+                try {
+                    await s3.putObject(params).promise();
+                    logger.info(`[workspace.js] synced ${bucketPath} to ${bucketName}`);
+                } catch (err) {
+                    logger.error(`[workspace.js] error syncing ${bucketPath}:\n ${err}`);
+                }
+            } else {
```

Comment:
Should we check whether this is a directory? As opposed to some other kind of weird file?